docs/visualize/vega.asciidoc

```diff
@@ -3,71 +3,2114 @@
-<<vega-useful-links, Resources and examples>>.
+Vega and Vega-Lite are capable of building most of the visualizations
+that {kib} provides, but with higher complexity. The most common reason
+to use Vega in {kib} is that your data or
+visualization isn't supported in other tools in {kib}, so you need
+to use the more powerful tool. The most common reasons to use Vega in {kib} are:
 
-[float]
-[[create-vega-viz]]
-=== Create Vega visualizations
+* Aggregations using the `nested` mapping
+* Aggregations with bucket scripts
+* Queries using that custom time filters
+* Complex calculations
+* Calculating percentage changes
+* Extracting data from _source instead of aggregation
+* Scatter charts
+* Sankey charts
+* Custom maps
+* Using a visual theme that {kib} does not provide
+
+Because Vega supports all of these, it is often easier to write a Vega spec
+than to develop a {kib} plugin when in need of more features.
+
+[[vega-lite-tutorial]]
+=== Tutorial: First visualization in Vega-Lite
 
-You create Vega visualizations by using the text editor, which is
-preconfigured with the options you need.
+When you first open the Vega editor in {kib}, you will see a pre-populated
+line chart which shows the total number of documents across all your indices
+within the time range.
 
 [role="screenshot"]
 image::images/vega_lite_default.png[]
 
-[float]
-[[vega-schema]]
-==== Change the Vega version
+This tutorial will show you how to build a stacked area chart, and how to debug
+the most common issues when writing a visualization in Vega:
 
-The default visualization uses Vega-Lite version 2. To use Vega version 4, edit
-the `schema`.
+[role="screenshot"]
+image::images/vega_lite_tutorial_1.png[]
 
-Go to `$schema`, enter `https://vega.github.io/schema/vega/v5.json`, then click
-*Update*.
+The text editor contains a Vega-Lite spec written in https://hjson.github.io/[HJSON],
+which is similar to JSON but optimized for human editing. HJSON supports:
 
-[float]
-[[vega-type]]
-==== Change the visualization type
+* Comments using // or /* syntax
+* Object keys without quotes
+* String values without quotes
+* Optional commas
+* Double or single quotes
 
-The default visualization is a line chart. To change the visualization type,
-change the `mark` value. The supported visualization types are listed in the
-text editor.
+Multiline strings are supported in HJSON, but will not work as expected in Vega.
 
-Go to `mark`, change the value to a different visualization type, then click
-*Update*.
+The default Vega-Lite spec is querying `_all` indices, but most visualizations are powered by a single index. Change the line:
+
+```yaml
+index: _all
+```
+
+to:
+
+```yaml
+index: kibana_sample_data_ecommerce
+```
+
+and click "Update". The result is probably not what you expect, because it shows
+a flat line with 0 results. This shows the first thing that you need to understand
+when writing a Vega spec in {kib}: small changes can lead to unexpected results.
```

Comment:
LOL :)  As a first lesson, I think we should rather add something like "baby steps - make sure to frequently refresh (i.e. Ctrl+Enter or auto-apply) - to see the result of each minor change". But I'm not too sure of how to phrase it, so maybe its ok to leave as is.